### PR TITLE
BAH 4142 Summary and Transformation

### PR DESCRIPTION
**Pull request by robfelder**
## Description
This PR covers migrating the following 4142 logic from the 526EZ flow:
- Summary of evidence page logic
- 4142 data transformation logic
- 4142 Schema refactoring (needed to move rules from uiSchema to local version of vets-json-schema schema)

## Testing done
- Manual & Unit

## Screenshots
screen shot 2018-11-20 at 3 38 10 pm

## Acceptance criteria
- [x] User provided facility names displays on the Summary of Evidence page
- [x] System successfully transforms the treatmentDateRange from an object to an array of objects
- [x] System successfully includes the `form4142` object with the submission payload

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by robfelder** [comment]
Good evening, I'm looking to get a code review done on this PR please. Have a great day!
@department-of-veterans-affairs/code-reviewers-for-external-teams-group
@rtluu

**Review by igreulich — APPROVED** [review]
lgtm

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 1]
```diff
@@ -179,6 +179,31 @@ export function transformRelatedDisabilities(object, claimedConditions) {
     .map(key => key.toLowerCase());
 }
 
+/**
+ * Converts the treatment date range into an array of objects from just an object
+ *
+ * @param {object} treatmentDateRange object containing from/to date range
+ * @returns {array} of treatmentDateRange's
+ */
+const transformDateRange = treatmentDateRange => [treatmentDateRange];
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/utils.jsx]: probably can just inline this e.g. `[facility.treatmentDateRate]`

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 2]
```diff
@@ -179,6 +179,31 @@ export function transformRelatedDisabilities(object, claimedConditions) {
...
+
+/**
+ * Cycles through the list of provider facilities and performs transformations on each property as needed
+ * @param {array} providerFacility array of objects being transformed
+ * @returns {object} containing the new Provider Facility structure
+ */
+const transformProviderFacilities = providerFacilities => {
+  const newProviderFacilities = providerFacilities.map(facility =>
+    _.set(
```
Comment (rianfowler) [inline src/applications/disability-benefits/all-claims/utils.jsx]: object spread is probably clearer here e.g. 
```
({
...facility,
treatmentDateRate: [facility.treatmentDateRate]
})
```

**Review by rianfowler — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 3]
```diff
@@ -179,6 +179,31 @@ export function transformRelatedDisabilities(object, claimedConditions) {
...
+ *
+ * @param {object} treatmentDateRange object containing from/to date range
+ * @returns {array} of treatmentDateRange's
+ */
+const transformDateRange = treatmentDateRange => [treatmentDateRange];
+
+/**
+ * Cycles through the list of provider facilities and performs transformations on each property as needed
+ * @param {array} providerFacility array of objects being transformed
+ * @returns {object} containing the new Provider Facility structure
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/utils.jsx]: I believe this is actually returning an array, not an object

File: src/applications/disability-benefits/all-claims/utils.jsx [hunk 4]
```diff
@@ -179,6 +179,31 @@ export function transformRelatedDisabilities(object, claimedConditions) {
...
+ */
+const transformProviderFacilities = providerFacilities => {
+  const newProviderFacilities = providerFacilities.map(facility =>
```
Comment (erikphansen) [inline src/applications/disability-benefits/all-claims/utils.jsx]: no need to store the mapped array in a const; you can just do `return providerFacilities.map(...`

**Comment by robfelder** [comment]
@erikphansen & @rianfowler - updates made, mind taking a quick peek?

**Review by erikphansen — APPROVED** [review]

**Pull request merged**